src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/UnusedImportsCheck.java

```diff
@@ -277,22 +277,36 @@ private boolean isUnusedImport(String imprt) {
     private void processIdent(DetailAST ast) {
         final DetailAST parent = ast.getParent();
         final int parentType = parent.getType();
+        final boolean isTypeDeclaration = TokenUtil.isTypeDeclaration(parentType);
```

Comment:
Please inline this in if statement like `if (TokenUtil.isTypeDeclaration(parentType))`, example above was only pseudocode.